packages/sdk/src/model/field/single-select.field.ts

```diff
@@ -1,5 +1,25 @@
-import { SingleSelectFieldCore } from '@teable/core';
+import { ColorUtils, SingleSelectFieldCore } from '@teable/core';
+import { keyBy } from 'lodash';
+import colors from 'tailwindcss/colors';
 import { Mixin } from 'ts-mixer';
 import { Field } from './field';
+import type { ISelectFieldDisplayChoice } from './multiple-select.field';
 
-export class SingleSelectField extends Mixin(SingleSelectFieldCore, Field) {}
+export class SingleSelectField extends Mixin(SingleSelectFieldCore, Field) {
+  private _choiceMap: Record<string, ISelectFieldDisplayChoice> = {};
```

Comment:
You don't have to repeat the code, create a new class SelectFieldSdk and put it in mixin